# DATA PREPARATION(THE NUMBERS)

## 【1】*Data Exploration*  

In [1]:
# Import neccesary libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

In [5]:
# Load the dataset
Number_df = pd.read_csv(r"C:\Users\JUDAH\Desktop\PHASE 2 PROJECT\GROUP-4_SK_-PHASE-2-PROJECT\ZippedData\tn.movie_budgets.csv.gz")
Number_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

## 【2】*Data Cleaning* 

In [6]:
# Check info
Number_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [7]:
# check shape
Number_df.shape

(5782, 6)

In [8]:
# Check for Duplicates
Number_df.duplicated().sum()

0

In [9]:
# Save the data
Number_df.to_csv('Cleaned_tn_movie_budgets.csv')

### *Data exploration and cleaning conclusion*

> The data from The numbers dataset is in quite mint condition with no missing values or Duplicated values, it has a shape of (5782, 6),showcasing rows and columns respectively.